client/router_create.go

```diff
@@ -109,34 +110,59 @@ func (cli *VanClient) GetVanControllerSpec(options types.SiteConfigSpec, van *ty
 	van.Controller.VolumeMounts = mounts
 	van.Controller.Sidecars = sidecars
 
-	serviceAccounts := []types.ServiceAccount{}
+	serviceAccounts := []*corev1.ServiceAccount{}
 	annotation := map[string]string{}
 	if options.AuthMode == string(types.ConsoleAuthModeOpenshift) {
 		annotation = map[string]string{
 			"serviceaccounts.openshift.io/oauth-redirectreference.primary": "{\"kind\":\"OAuthRedirectReference\",\"apiVersion\":\"v1\",\"reference\":{\"kind\":\"Route\",\"name\":\"skupper-controller\"}}",
 		}
 	}
-	serviceAccounts = append(serviceAccounts, types.ServiceAccount{
-		ServiceAccount: "skupper-proxy-controller",
-		Annotations:    annotation,
+	serviceAccounts = append(serviceAccounts, &corev1.ServiceAccount{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "ServiceAccount",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        "skupper-proxy-controller",
+			Annotations: annotation,
+		},
```

Comment:
Just a thought, I wonder if it would be worth having a method in the respective kube packages to create a basic object of each type? It might reduce some of the boiler plate, especially for commonly used types. Not sure though, it is just a suggestion.